docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,454 @@
+
+The actual transfer of transactions happens through the `sendTxs()` method. It then sends the transactions as bytes to the `avalanchego` service through `appSender` that is being provided to VM during initialization. The `avalanchego` then gossips the transactions to the subnet's validators.
+
+```go title="/blobvm/vm/network.go"
+func (n *PushNetwork) sendTxs(txs []*chain.Transaction) error {
+    if err := n.vm.appSender.SendAppGossip(b); err != nil {
+        log.Warn(
+            "GossipTxs failed",
+            "error", err,
+        )
+        return err
+    }
+}
+```
+
+The other validators similarly receive incoming transactions from other validators via the `avalanchego` service on `vm/network.go`'s `AppGossip()` method. Once transaction bytes are received, it submits unmarshalled transaction objects to the VM.
+
+```go
+func (vm *VM) AppGossip(nodeID ids.NodeID, msg []byte) error {
+	txs := make([]*chain.Transaction, 0)
+	if _, err := chain.Unmarshal(msg, &txs); err != nil {
+		return nil
+	}
+
+	// submit incoming gossip
+	log.Debug("AppGossip transactions are being submitted", "txs", len(txs))
+	if errs := vm.Submit(txs...); len(errs) > 0 {
+		for _, err := range errs {
+
+		}
+	}
+    
+	return nil
+}
+```
+
+### Service
+
+[Service](https://github.com/ava-labs/blobvm/blob/master/vm/public_service.go) implements the API handlers to access the functions of the VM. VM has a method called [`CreateHandlers()`](https://github.com/ava-labs/blobvm/blob/1fbb655246a2a41861f40d603ad40299b177252a/vm/vm.go#L265) that will return the HTTP handler.
+
+```go
+func (vm *VM) CreateHandlers() (map[string]*common.HTTPHandler, error) {
+    apis := map[string]*common.HTTPHandler{}
+    public, err := newHandler(Name, &PublicService{vm: vm})
+    if err != nil {
+        return nil, err
+    }
+    apis[PublicEndpoint] = public
+    return apis, nil
+}
+```
+
+In the above function, a new RPC server is created with the `PublicService`. It implements all the RPCs supported by the virtual machine. It has access to a VM instance in its structure, that is passed while creating it (see the above function).
+
+```go
+type PublicService struct {
+	vm *VM
+}
+```
+
+Let's have a look at the [`IssueRawTx()`](https://github.com/ava-labs/blobvm/blob/1fbb655246a2a41861f40d603ad40299b177252a/vm/public_service.go#L63) service method -
+
+```go
+func (svc *PublicService) IssueRawTx(_ *http.Request, args *IssueRawTxArgs, reply *IssueRawTxReply) error {
+	tx := new(chain.Transaction)
+	if _, err := chain.Unmarshal(args.Tx, tx); err != nil {
+		return err
+	}
+
+	// otherwise, unexported tx.id field is empty
+	if err := tx.Init(svc.vm.genesis); err != nil {
+		return err
+	}
+	reply.TxID = tx.ID()
+
+	errs := svc.vm.Submit(tx)
+	if len(errs) == 0 {
+		return nil
+	}
+	if len(errs) == 1 {
+		return errs[0]
+	}
+	return fmt.Errorf("%v", errs)
+}
+```
```

Comment:
[Fenced code blocks should be surrounded by blank lines](https://github.com/DavidAnson/markdownlint/blob/v0.25.1/doc/Rules.md#md031)